hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/ProbeUtils.java

```diff
@@ -16,122 +16,95 @@
 
 package com.hazelcast.internal.metrics.impl;
 
+import static com.hazelcast.internal.util.MapUtil.createHashMap;
+import static java.util.Collections.unmodifiableMap;
+
 import com.hazelcast.internal.util.counters.Counter;
 
-import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
-import java.util.List;
+import java.util.LinkedHashSet;
 import java.util.Map;
+import java.util.Objects;
 import java.util.concurrent.Semaphore;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.LongAccumulator;
 import java.util.concurrent.atomic.LongAdder;
-import java.util.stream.Stream;
-
-import static com.hazelcast.internal.util.MapUtil.createHashMap;
-import static java.util.Collections.unmodifiableMap;
 
 /**
  * Utility functions for probes.
  */
-final class ProbeUtils {
-
-    static final int TYPE_PRIMITIVE_LONG = 1;
-    static final int TYPE_LONG_NUMBER = 2;
-
-    static final int TYPE_DOUBLE_PRIMITIVE = 3;
-    static final int TYPE_DOUBLE_NUMBER = 4;
-
-    static final int TYPE_COLLECTION = 5;
-    static final int TYPE_MAP = 6;
-    static final int TYPE_COUNTER = 7;
-    static final int TYPE_SEMAPHORE = 8;
-
-    private static final Map<Class<?>, Integer> TYPES;
+enum ProbeUtils {
```

Comment:
There are 2 concerns conflated:

1) A utility class for probes
2) An enum for the different types of probes

I would at least rename the ProbeUtils to ProbeType or something so that you can pass a ProbeType as a parameter instead of passing ProbeUtils as a parameter.

Reply:
Good point, done.

fb5db6a8dd6266ffebb5691cf612f5ec64d68276
5fccd6ffd5ba28b26fb01d44b14cdafa90e4b8ba